pkg/virt-handler/mount-manager/utils/recorder.go

```diff
@@ -0,0 +1,296 @@
+package recorder
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+	v1 "kubevirt.io/api/core/v1"
+
+	diskutils "kubevirt.io/kubevirt/pkg/ephemeral-disk-utils"
+	"kubevirt.io/kubevirt/pkg/safepath"
+	"kubevirt.io/kubevirt/pkg/unsafepath"
+)
+
+//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
+
+type HotpluggedDisksMountTargetEntry struct {
+	TargetFile string `json:"targetFile"`
+}
+
+type ContainerDisksMountTargetEntry struct {
+	TargetFile string `json:"targetFile"`
+	SocketFile string `json:"socketFile"`
```

Comment:
Why do we have separate `MountTargetEntry` for hotplug and container disk. The fields are the same, except that container disks have an extra `SocketFile`, would it make sense to just have a singular `MountTargetEntry` with an optional `SocketFile` and simply return all the entries and not distinguish between the two types?

I think that would simplify a lot of the code here since we don't have to duplicate a bunch of stuff for both types.

Reply:
That's a good point! I'll try this

Reply:
@awels thinking a bit more about it. I think we still need to make the distinguish. The main reason is that container disks and hotplug can be deleted at different times. Hotplugged volumes could be added and removed at runtime while container disks only at start-up and clean-up time.If we would have other mouts time,then it would be probably make sense only to categorize them with immutable and mutable. Does it make sense to you?

Reply:
So AFAIK the key is the target file, and it doesn't matter when we want to unmount the file. We can find the entry by the key. So if we unplug a hotplugged volume, the code that uses that will look up the entry by target file, and it will find the right one if it exists in the cache/entry file. The socket file is just an optional attribute, and only relevant for when the container disk code runs to unmount.

Reply:
But now the hotplugged volumes can be unmounted also at runtime not only on the VM clean-up (see: https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/vm.go#L2801-L2806). If I unify the two mount entries and don't make the difference between container disks and hotplug, I cannot unmount the hotplugged disks at runtime anymore. 
I would be a bit reluctant in unmounting the hotplugged disks only on clean-up.

Reply:
I don't understand why you can't unmount entries at runtime if the entries are unified in the cache/file. They are still under the same VMI, the list might just be some entries larger when searching.

Reply:
I might have misunderstood your comment. Are you suggesting simply to rename and use a single entry type like this:
```golang
type MountTargetEntry struct {
        TargetFile string `json:"targetFile"`
        SocketFile string `json:"socketFile,omitempty"`
}       

type VMIMountTargetEntry struct {
        HotpluggedVolumes []MountTargetEntry `json:"hotpluggedDisks"`
        ContainerDisks    []MountTargetEntry `json:"containerDisks"`
}
```
or to put together container disks and hotplugged disks like: 
```golang 
type MountTargetEntry struct {
        TargetFile string `json:"targetFile"`
        SocketFile string `json:"socketFile,omitempty"`
} 
type VMIMountTargetEntry struct {
        MountedVolumes []MountTargetEntry `json:"mountedVolumes"`
}
```
The second option doesn't allow me to distinguish between removing the mount points at runtime and only at clean-up

Reply:
I was thinking option 2 as it is a logical extension of option 1

Why are we interested in distinguishing between runtime, and only at clean-up. We just want to look up the entries in the file/cache, and do something based on that. I am not sure it is relevant when the lookup happens. I guess logically speaking you should only remove container disk mounts at cleanup, but that logic should be part of the container cleanup, not the entry logic.

Reply:
I'm worried that deleting the hotplugged disk will also remove the entries for the container disks. The functions [DeleteContainerDisksMountRecord](https://github.com/alicefr/kubevirt/blob/862b90e59c594ad4625aa45b08e80fc2f64bd146/pkg/virt-handler/mount-manager/utils/recorder.go#L116) and [DeleteHotpluggedVolumesMountRecord](https://github.com/alicefr/kubevirt/blob/862b90e59c594ad4625aa45b08e80fc2f64bd146/pkg/virt-handler/mount-manager/utils/recorder.go#L120) could be call at different moments